tartufo/scanner.py

```diff
@@ -476,3 +477,52 @@ def chunks(self):
         )
         for blob, file_path in self._iter_diff_index(diff_index):
             yield types.Chunk(blob, file_path)
+
+
+class FolderScanner(ScannerBase):
+    """Used to scan a folder."""
+
+    folder_options: types.FolderOptions
+    folder_path: str
+
+    def __init__(
+        self,
+        global_options: types.GlobalOptions,
+        folder_options: types.FolderOptions,
+        folder_path: str,
+    ) -> None:
+        """Used for scanning a folder.
+
+        :param global_options: The options provided to the top-level tartufo command
+        :param folder_options: The options specific to interacting with a folder
+        :param folder_path: The local filesystem path to scan
+        """
+        self.folder_options = folder_options
+        self.folder_path = folder_path
+        super().__init__(global_options)
+
+    @property
+    def chunks(self):
+        """Yield the individual file changes currently staged for commit.
+
+        :rtype: Generator[Chunk, None, None]
+        """
+
+        for blob, file_path in self._iter_folder():
+            if self.should_scan(file_path):
+                yield types.Chunk(blob, file_path)
+
+    def _iter_folder(self):
```

Comment:
This needs a return annotation. I think it should also accept an argument of the path to be scanned, following the pattern of the generators on the other scanners.

Reply:
Why would we pass the path in if the path is passed into the constructor and the `_iter_folder` is an internal method to the class? If this needs to be reused internally, cant it be refactored then? Seems more of a personal preference.

Reply:
I would agree that instance methods don't need to take member fields as parameters.